sklearn/tests/test_discriminant_analysis.py

```diff
@@ -329,6 +359,51 @@ def test_lda_store_covariance():
     )
 
 
+def test_lda_shrinkage():
+    """
+    Test that shrunk covariance estimator and
+    shrinkage parameter behave the same
+    """
+    X = np.random.rand(100, 10)
+    Y = np.random.randint(3, size=(100))
+    c1 = LinearDiscriminantAnalysis(store_covariance=True, shrinkage=0.5,
+                                    solver="lsqr")
+    c2 = LinearDiscriminantAnalysis(
+            store_covariance=True,
+            covariance_estimator=ShrunkCovariance(shrinkage=0.5),
+            solver="lsqr")
+    c1.fit(X, Y)
+    c2.fit(X, Y)
+    for i in range(2):
+        assert np.allclose(c1.means_[i], c2.means_[i])
+    assert np.allclose(c1.covariance_, c2.covariance_)
+
+
+def test_lda_ledoitwolf():
+    """
+    When shrinkage="auto" current implementation
+    uses ledoitwolf estimation of
+    covariance after standardizing the data.
+    This checks that it is indeed the case
+    """
+    X = np.random.rand(100, 10)
+    Y = np.random.randint(3, size=(100,))
+    c1 = LinearDiscriminantAnalysis(
+            store_covariance=True,
+            shrinkage="auto",
+            solver="lsqr")
+    c2 = LinearDiscriminantAnalysis(
+            store_covariance=True,
+            covariance_estimator=LedoitWolf(),
+            solver="lsqr",
+            standardize=True)
+    c1.fit(X, Y)
+    c2.fit(X, Y)
+    for i in range(2):
+        assert np.allclose(c1.means_[i], c2.means_[i])
+    assert np.allclose(c1.covariance_, c2.covariance_)
```

Comment:
Use the `assert_allclose` function

Reply:
I think we prefer using `np.testing.assert_allclose`  now